include/itkImageToTorchImageFilter.h

```diff
@@ -19,60 +19,88 @@
 #define itkImageToTorchImageFilter_h
 
 #include "itkImageToImageFilter.h"
+#include "itkImageRegionIterator.h"
+#include "itkImageRegionConstIterator.h"
+
+#include "itkTorchImage.h"
 
 namespace itk
 {
 
 /** \class ImageToTorchImageFilter
  *
- * \brief Filters a image by iterating over its pixels.
+ * \brief Converts an itk::Image to an itk::TorchImage.
  *
- * Filters a image by iterating over its pixels in a multi-threaded way
- * and {to be completed by the developer}.
+ * Conversions between Image and TorchImage permit pipelines with
+ * steps for either kind of image.
  *
  * \ingroup PyTorch
  *
  */
-template <typename TInputImage, typename TOutputImage>
-class ImageToTorchImageFilter : public ImageToImageFilter<TInputImage, TOutputImage>
+template< typename TInputImage, typename TOutputTorchImage >
+class ITK_TEMPLATE_EXPORT ImageToTorchImageFilter : public ImageToImageFilter< TInputImage, TOutputTorchImage >
 {
 public:
-  ITK_DISALLOW_COPY_AND_ASSIGN(ImageToTorchImageFilter);
+  static constexpr bool ConceptCheck =
+    std::is_same< TInputImage, Image< typename TInputImage::PixelType, TInputImage::ImageDimension > >::value &&
+    std::is_same< TOutputTorchImage, TorchImage< typename TOutputTorchImage::PixelType, TOutputTorchImage::ImageDimension > >::value &&
```

Comment:
```suggestion
std::is_same< TOutputTorchImage, TorchImage< typename TInputImage::PixelType, TInputImage::ImageDimension > >::value &&
```
With this change it is redundant with checks on lines 47 and 48. Without this change it is tautological too.

Reply:
I will make it simpler, enabled by your suggestion to drop the second template parameter.